src/main/webapp/app/shared/notification/notification-popup/notification-popup.component.ts

```diff
@@ -0,0 +1,106 @@
+import { Router, UrlTree } from '@angular/router';
+import { NotificationService } from 'app/shared/notification/notification.service';
+import { User } from 'app/core/user/user.model';
+import { AccountService } from 'app/core/auth/account.service';
+import { Notification } from 'app/entities/notification.model';
+import { GroupNotification } from 'app/entities/group-notification.model';
+
+@Component({
+    selector: 'jhi-notification-popup',
+    templateUrl: './notification-popup.component.html',
+    styleUrls: ['./notification-popup.scss'],
+})
+export class NotificationPopupComponent implements OnInit {
+    notifications: Notification[] = [];
+
+    constructor(private accountService: AccountService, private notificationService: NotificationService, private router: Router) {}
+
+    /**
+     * Subscribe to notification updates that are received via websocket if the user is logged in.
+     */
+    ngOnInit(): void {
+        this.accountService.getAuthenticationState().subscribe((user: User | undefined) => {
+            if (user) {
+                this.subscribeToNotificationUpdates();
+            }
+        });
+    }
+
+    /**
+     * Returns a string that can be interpreted as fontawesome icon based on the notification type of the given notification.
+     * @param notification {Notification}
+     */
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    notificationIcon(notification: Notification): string {
+        return 'check-double';
+    }
+
+    /**
+     * Removes the notification at the specified index from the notifications array.
+     * @param index {number}
+     */
+    removeNotification(index: number): void {
+        this.notifications.splice(index, 1);
+    }
+
+    /**
+     * Navigate to the target (view) of the notification that the user clicked.
+     * @param notification {Notification}
+     */
+    navigateToTarget(notification: Notification): void {
+        this.router.navigateByUrl(this.notificationTargetRoute(notification));
+    }
+
+    private notificationTargetRoute(notification: Notification): UrlTree | string {
+        if (notification.target) {
+            const target = JSON.parse(notification.target);
+            return this.router.createUrlTree([target.mainPage, target.course, target.entity, target.id]);
+        }
+        return this.router.url;
+    }
+
+    private subscribeToNotificationUpdates(): void {
+        this.notificationService.subscribeToNotificationUpdates().subscribe((notification: Notification) => {
+            this.addNotification(notification);
+        });
+    }
+
+    private addNotification(notification: Notification): void {
+        // Only add a notification if it does not already exist.
+        if (notification && !this.notifications.some(({ id }) => id === notification.id)) {
+            // For now only notifications about a started quiz should be displayed.
+            if (notification.title === 'Quiz started') {
```

Comment:
This seems very brittle. Can this be changed to an enum?

Reply:
We will have dedicated notification types in the future. So I will adjust it then :-)